File: macos/Scanner/Sources/Scanner/Detectors/ClaudeDesktopDetector.swift
public struct ClaudeDesktopDetector: Detector {
    public let id = "claude-desktop"
    public let displayName = "Claude Desktop"

    public init() {}

    public func collectFacts(fs: any FilesystemFacade) throws -> ScanFacts {
        let configPath = homePath(fs, "Library/Application Support/Claude/claude_desktop_config.json")
        var facts = ScanFacts()
        appendAppInstallationFact(
            &facts,
            fs: fs,
            appId: id,
            candidatePaths: ["/Applications/Claude.app", homePath(fs, "Applications/Claude.app")]
        )

        if let text = try readIfPresent(fs, path: configPath) {
            facts.mcpServers.append(contentsOf: try FactParsers.parseMcpServersJson(text, appId: id, configPath: configPath))
            appendConfigFileFact(&facts, fs: fs, appId: id, path: configPath)
        }

        return facts
    }
}

File: windows/Scanner/Detectors/ClaudeDesktopDetector.cs
using AIExposureScanner.Scanner.Filesystem;
using AIExposureScanner.Scanner.Models;
using AIExposureScanner.Scanner.Parsing;

namespace AIExposureScanner.Scanner.Detectors;

public sealed class ClaudeDesktopDetector : IDetector
{
    public string Id => "claude-desktop";
    public string DisplayName => "Claude Desktop";

    public ScanFacts CollectFacts(IFilesystemFacade fs)
    {
        var configPath = Path.Combine(fs.ApplicationDataDirectory, "Claude", "claude_desktop_config.json");

        var facts = new ScanFacts();
        DetectorSupport.AppendAppInstallationFact(
            facts,
            fs,
            Id,
            [Path.Combine(fs.LocalApplicationDataDirectory, "AnthropicClaude")]
        );

        var text = DetectorSupport.ReadIfPresent(fs, configPath);
        if (text is not null)
        {
            facts.McpServers.AddRange(FactParsers.ParseMcpServersJson(text, Id, configPath));
            DetectorSupport.AppendConfigFileFact(facts, fs, Id, configPath);
        }
        return facts;
    }
}
